examples/js/postprocessing/EffectComposer.js

```diff
@@ -91,6 +93,28 @@ Object.assign( THREE.EffectComposer.prototype, {
 
 		var maskActive = false;
 
+		var currentVREnabled = this.renderer.vr.enabled;
+
+		if ( this.renderer.vr.enabled === true ) {
+
+			this.renderer.vr.enabled = false;
+
+			var isPresenting = this.renderer.vr.isPresenting();
+
+			// renderer size can be changed by WebXR/VRManager
+			// on presenting change
+
+			if ( isPresenting !== this._wasPresenting ) {
+
+				this.renderer.getDrawingBufferSize( this._rendererSize );
+				this.setSize( this._rendererSize.x, this._rendererSize.y );
+
+			}
+
+			this._wasPresenting = isPresenting;
```

Comment:
If we can detect presenting change, we can move this code out from `.render()` like this.

```javascript
var renderSize = new THREE.Vector2();

window.addEventListener( 'vrdisplaypresentchange', function ( event ) {

    renderer.getDrawingBufferSize( rendererSize );
    composer.setSize( rendererSize.x, rendererSize.y );   

}, false );
```

This is with WebVR API. Can we similarly do with WebXR API?

Reply:
maybe devicechanged? https://www.w3.org/TR/webxr/#event-types

Reply:
Opened #16277

Reply:
`devicechange` doesn't look an event for present change, but for device availability change.

Reply:
Now WebXR/VRManager trigger present change event and I use.